rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_node.hpp

```diff
@@ -268,10 +269,14 @@ class LifecycleNode : public node_interfaces::LifecycleNodeInterface,
 
   /// Create and return a Service.
   /**
-   * \sa rclcpp::Node::create_service
+   * \param[in] service_name The topic to service on.
+   * \param[in] callback User-defined callback function.
+   * \param[in] qos_profile rmw_qos_profile_t Quality of service profile for client.
+   * \param[in] group Callback group to call the service.
+   * \return Shared pointer to the created lifecycle service.
    */
   template<typename ServiceT, typename CallbackT>
-  typename rclcpp::Service<ServiceT>::SharedPtr
+  typename rclcpp_lifecycle::LifecycleService<ServiceT>::SharedPtr
```

Comment:
I know this is the same we do for publishers, but IMO it would be better to have `create_service()`/`create_publisher()` alongside `create_lifecycle_service()`/`create_lifecycle_publisher()`.

You can currently still create a non-managed entity by using the full method path, but having non-overriding methods would be clearer IMO.

---

This would be a breaking change, so we can ignore it for the moment.

Reply:
I'm not sure. Consider some code that operates on a node-like object. It nice to be able to pass in an `rclcpp::Node` or `rclcpp_lifecycle::LifecycleNode` to the same API (provided both classes are implementing the required interface). In this scenario, if we have lifecycle-specific methods, we'd end up not using the managed entities for lifecycle.

However, I'm not sure if this idea of passing Node and LifecycleNode interchangeable is true in practice.

If a user chooses LifecycleNode, then I would think they want managed entities in most cases. So, I lean towards  "overriding" `create_service()`, `create_publisher()`, etc. If the user wants to create an non-managed entity, they still have that option.

I guess either way, we'll be breaking API (currently this PR is changing the return type).

Reply:
_Coming back to this a bit later being interested in the suite of managed entity implementation_

I would agree with @ivanpauno thinking from a potential user bug perspective. 

I could imagine a user who wants a mixture of entity types with some being active even in `Inactive` state (e.g., a `GetXParam.srv` used to grab a loaded variable while the node is `Inactive` as you don't want to spin other entities). 

This may* be poor user design in theory but I could easily see someone trying to do this. If `create_service` was overriden to default to a managed entity, it would be difficult to explicitly see why the service is turned off/not accepting requests as it was automatically turned into a `ManagedEntity` type under the hood.

@jacobperron I think you bring up a great point, I'm not sure the feature parity between the two at the moment

Thinking out loud: I can see the other side, however, where you likely want to use `ManagedEntity` types and discourage any non-managed entity types within lifecycle nodes. Possibly there could be a compiler warning or equivalent when calling `create_service` instead of `create_lifecycle_service` from within a lifecycle node.